source/entries.d.ts

```diff
@@ -0,0 +1,85 @@
+// `Map` Helpers
+type MapKey<BaseType> = BaseType extends Map<infer KeyType, unknown> ? KeyType : never;
+type MapValue<BaseType> = BaseType extends Map<unknown, infer KeyType> ? KeyType : never;
+
+// `Entry` Helpers
+type ArrayEntry<BaseType extends readonly unknown[]> = [number, BaseType[number]];
+type MapEntry<BaseType> = [MapKey<BaseType>, MapValue<BaseType>];
+type ObjectEntry<BaseType> = [keyof BaseType, BaseType[keyof BaseType]];
+type SetEntry<BaseType> = BaseType extends Set<infer ItemType> ? [ItemType, ItemType] : never;
+
+/**
+Many collections have a `.entries()` method that will return an enumerable array of that structure's keys and values.  The `Entry` type will return the type of that collection's entry give the type itself.
+
+For example the {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries|`Object`}, {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/entries|`Map`}, {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/entries|`Array`}, and {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/entries|`Set`}
+collections all have this method.  Note that since `WeakMap` and `WeakSet` do not have this method because they are not enumerable.
```

Comment:
```suggestion
collections all have this method. Note that since `WeakMap` and `WeakSet` do not have this method since their entries are not enumerable.
```

Reply:
thanks! fixed in https://github.com/sindresorhus/type-fest/pull/130/commits/1362c622d84f76445c2246389e671979dd8e65e9